test/integration/listener_lds_integration_test.cc

```diff
@@ -1582,6 +1582,7 @@ TEST_P(ListenerFilterIntegrationTest,
   ASSERT_TRUE(fake_upstreams_[0]->waitForRawConnection(fake_upstream_connection));
   ASSERT_TRUE(fake_upstream_connection->waitForData(data.size(), &data));
   tcp_client->close();
+  ASSERT_TRUE(fake_upstream_connection->waitForDisconnect());
```

Comment:
emm...I need to think about how to avoid this happened again, people may invoke EXPECT_LOG_CONTAINS before upstream connection disconnect in other place

Reply:
I don't have anything in my mind, since it feels expensive to add lock for `Envoy::Logger::DelegatingLogSink::setShouldEscape(bool)`